pkg/cosign/github.go

```diff
@@ -17,12 +17,22 @@ type GitHubPublicKey struct {
 func getGitHubPublicKeys(username string) ([]GitHubPublicKey, error) {
 	logrus.Debugf("Getting public keys for GitHub user %s", username)
 	url := fmt.Sprintf("https://api.github.com/users/%s/keys", username)
-	resp, err := http.Get(url)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Accept", "application/json")
+	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
```

Comment:
No error handling on this read? Not sure if that would cause issues since we are trying to include the body in the error message. Also since we use the body whether we're erroring out or continuing the function we could refactor this body read out of the if before the check maybe

Reply:
`io.ReadAll` always returns a byte slice, even on failure, so I think this should not error out.
You're right though, that we could refactor this out since we use the body anyway, but I thought it might look weird since we do care about the read error if we need to use the body... within the status code handling, the body is just a nice to have :man_shrugging: